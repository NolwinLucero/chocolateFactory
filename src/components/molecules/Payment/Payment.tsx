import React, { useState } from 'react';
import styled from 'styled-components';

interface Props {
  credits: number;
  onCreditChangeHandler: (credit: string) => void;
}

const validPayments = ['10c', '20c', '50c', '$1', '$2'];

export const Payment: React.FC<Props> = ({
  credits,
  onCreditChangeHandler,
}) => {
  const [selectedCoin, setSelectedCoin] = useState('');

  const onClickHandler = (e: React.FormEvent) => {
    e.preventDefault();

    if (validPayments.indexOf(selectedCoin) < 0) {
      alert('Invalid amount!');
      return;
    }

    if (selectedCoin) {
      onCreditChangeHandler(selectedCoin);
    }
  };

  const onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    const val = e.target.value;
    setSelectedCoin(val);
  };

  return (
    <>
      <h1 data-testid="credits">Credits: ${credits?.toFixed(2) || 0}</h1>
      <StyledPaymentDiv>
        <form>
          <label>
            Input coin:
            <input
              data-testid="credit"
              type="text"
              onChange={onChangeHandler}
            />
          </label>
          <button data-testid="addCredit" onClick={onClickHandler}>
            Drop
          </button>
        </form>
      </StyledPaymentDiv>
    </>
  );
};

const StyledPaymentDiv = styled.div``;
