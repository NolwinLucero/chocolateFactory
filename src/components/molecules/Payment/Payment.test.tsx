import React from 'react';
import ReactDOM from 'react-dom';
import { Payment } from './Payment';
import renderer from 'react-test-renderer';
import { render, cleanup, fireEvent } from '@testing-library/react';

const credits = 5;
const onCreditChangeHandler = () => {
  alert('clicked');
};

window.alert = jest.fn();

afterEach(cleanup);
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Payment
      credits={credits}
      onCreditChangeHandler={onCreditChangeHandler}
    ></Payment>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders credits', () => {
  const { getByTestId } = render(
    <Payment
      credits={credits}
      onCreditChangeHandler={onCreditChangeHandler}
    ></Payment>
  );
  expect(getByTestId('credits')).toHaveTextContent('5');
});

it('add credits button', () => {
  const { getByTestId } = render(
    <Payment
      credits={credits}
      onCreditChangeHandler={onCreditChangeHandler}
    ></Payment>
  );

  const button = getByTestId('addCredit');
  fireEvent.click(button);
});

it('change credit', () => {
  const { getByTestId } = render(
    <Payment
      credits={credits}
      onCreditChangeHandler={onCreditChangeHandler}
    ></Payment>
  );

  const input = getByTestId('credit');
  fireEvent.change(input, { target: { value: '$1' } });
});

it('matches snapshot', () => {
  const tree = renderer
    .create(
      <Payment
        credits={credits}
        onCreditChangeHandler={onCreditChangeHandler}
      ></Payment>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
