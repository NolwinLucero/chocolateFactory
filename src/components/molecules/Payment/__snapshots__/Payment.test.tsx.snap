// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <h1
    data-testid="credits"
  >
    Credits: $
    5.00
  </h1>,
  <div
    className="sc-bdvvtL"
  >
    <form>
      <label>
        Input coin:
        <input
          data-testid="credit"
          onChange={[Function]}
          type="text"
        />
      </label>
      <button
        data-testid="addCredit"
        onClick={[Function]}
      >
        Drop
      </button>
    </form>
  </div>,
]
`;
